### PR TITLE
FIX: ImageViewer uses proxy address when present

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Issue

ImageViewer does not show images when the network uses proxy.

##### Description of changes

Widget read the proxy address from http_response and use it when retrieving the images. 

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by PrimozGodec** [comment]
@ales-erjavec can you please look at PR?

File: orangecontrib/imageanalytics/widgets/owimageviewer.py [hunk 1]
```diff
@@ -1190,6 +1190,16 @@ def __init__(self, parent=None):
             )
             netmanager.setCache(cache)
             ImageLoader._NETMANAGER_REF = weakref.ref(netmanager)
+
+            proxy_address = os.environ.get('http_proxy')
```
Comment (lanzagar) [inline orangecontrib/imageanalytics/widgets/owimageviewer.py]: This reads (and sets?) just the `http_proxy`, what about `https_proxy`?

File: orangecontrib/imageanalytics/widgets/owimageviewer.py [hunk 2]
```diff
@@ -1190,6 +1190,16 @@ def __init__(self, parent=None):
...
+            if proxy_address is not None:
+                ip_port = proxy_address.split(":")
+                ip, port = ip_port \
+                    if len(ip_port) == 2 else (ip_port[0], '8080')
+                proxy = QNetworkProxy(
+                    QNetworkProxy.HttpProxy, ip, int(port))
+                netmanager.setProxy(proxy)
+
```
Comment (ales-erjavec) [inline orangecontrib/imageanalytics/widgets/owimageviewer.py]: Maybe try this:
```diff
@@ -31,7 +31,8 @@ from AnyQt.QtCore import (
 )
 from AnyQt.QtCore import pyqtSignal as Signal, pyqtSlot as Slot
 from AnyQt.QtNetwork import (
-    QNetworkAccessManager, QNetworkDiskCache, QNetworkRequest, QNetworkReply
+    QNetworkAccessManager, QNetworkDiskCache, QNetworkRequest, QNetworkReply,
+    QNetworkProxyFactory, QNetworkProxyQuery
 )
 
 import Orange.data
@@ -1170,6 +1171,11 @@ class OWImageViewer(widget.OWWidget):
         self.clear()
 
 
+class SystemProxyFactory(QNetworkProxyFactory):
+    def queryProxy(self, query=QNetworkProxyQuery()):
+        return self.systemProxyForQuery(query)
+
+
 class ImageLoader(QObject):
     #: A weakref to a QNetworkAccessManager used for image retrieval.
     #: (we can only have one QNetworkDiskCache opened on the same
@@ -1189,6 +1195,7 @@ class ImageLoader(QObject):
                              __name__ + ".ImageLoader.Cache")
             )
             netmanager.setCache(cache)
+            netmanager.setProxyFactory(SystemProxyFactory())
             ImageLoader._NETMANAGER_REF = weakref.ref(netmanager)
         self._netmanager = netmanager
```

At the least do not parse http_proxy like that. For instance that fails for ```http://user:1234@example.com:8000``` Use `QUrl` or `urlparse`.

Comment (PrimozGodec) [reply orangecontrib/imageanalytics/widgets/owimageviewer.py]: @ales-erjavec thank you for the suggestion. I tried this implementation on our WIFI which uses the proxy (Orange 5G, proxy IP is: 192.168.0.1:3128), but cannot make it work. Do you have an idea what to try or how to debug it?

Comment (ales-erjavec) [reply orangecontrib/imageanalytics/widgets/owimageviewer.py]: Works for me on macos 10.11 with Qt 5.9.3 and system configured proxy.

Can you try https://github.com/ales-erjavec/orange3-imageanalytics/tree/proxy and configure via the orange preferences dialog.

Comment (PrimozGodec) [reply orangecontrib/imageanalytics/widgets/owimageviewer.py]: Work for me too #94

**Pull request closed without merging**